src/sinks/amqp/service.rs

```diff
@@ -128,11 +126,8 @@ impl Service<AmqpRequest> for AmqpService {
             match fut {
                 Ok(result) => match result.await {
                     Ok(lapin::publisher_confirm::Confirmation::Nack(_)) => {
-                        warn!("Received Negative Acknowledgement from AMQP server.");
-                        Ok(AmqpResponse {
-                            json_size: req.metadata.into_events_estimated_json_encoded_byte_size(),
-                            byte_size,
-                        })
+                        emit!(AmqpNackError);
+                        Err(AmqpError::AmqpNack)
```

Comment:
A negative acknowledgement should be treated as a failure.

Reply:
Should we document that somewhere? Either a code comment here or in a design spec location.

Reply:
I would argue that emitting/returning an error here is self-documenting.  To be clear, this change is simply improving the observability of this component, giving the user insight into the number of Nacks that occur.